### PR TITLE
Improve versioning

### PR DESCRIPTION
**Pull request by balopat**
This is regarding #2775.

- hack/new_version now checks via the Github API whether the current version is released or not - if not then fails 
- hack/new_version.sh now takes the new version as an argument for easier testing, e.g, try it out with `hack/new_version.sh v1beta15`!

next PRs will cater for :
- add a PR check that uses this release logic to ensure that no released version is mutated

### CODE REVIEW

**Comment by briandealwis** [comment]
Could the `release.sh` script compare `.../latest/config.go` — or the generated schema? — since the last tagged release?

The last tagged release is available from `git describe --match 'v[0-9]*' --abbrev 0` ([source](https://stackoverflow.com/questions/3867619/how-to-get-last-git-tag-matching-regex-criteria#comment26823639_3867811)).

**Comment by balopat** [comment]
> I'm not sure I like the usage of an alias for `latest`. We've been there and it's harder to use. For example, you don't get auto import from the IDE when you write `latest.SkaffoldConfig{}`

The autoimport is a good point. I think the `latest` pointer is not a vital thing, I can introduce the `Released` flag without it. I'll split out the timeout increase!

**Comment by balopat** [comment]
> Could the `release.sh` script compare `.../latest/config.go` — or the generated schema? — since the last tagged release?
> 
> The last tagged release is available from `git describe --match 'v[0-9]*' --abbrev 0` ([source](https://stackoverflow.com/questions/3867619/how-to-get-last-git-tag-matching-regex-criteria#comment26823639_3867811)).

That could work too, thanks for the suggestion! 
Pros: it can't be spoofed + no need for an extra bool flag. 
Cons: now the contributor won't have a visual confirmation before trying to add a new field - but we can  still add a comment before the config version though in the release script.

File: hack/new_version.sh [hunk 1]
```diff
@@ -14,11 +14,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-echo "Current skaffold config version: " $(sed -n 's;.*Version.*"skaffold/\(.*\)";\1;p' pkg/skaffold/schema/latest/config.go)
```
Comment (dgageot) [inline hack/new_version.sh]: Isn't this message useful when a version is not provided as an argument?

File: hack/new_version.sh [hunk 2]
```diff
@@ -14,11 +14,14 @@
...
+set -e
 
-echo
+NEW_VERSION=$1
+
+if [[ "" == "$NEW_VERSION" ]]; then
```
Comment (dgageot) [inline hack/new_version.sh]: `if [[ -z "$NEW_VERSION" ]]`

File: hack/new_config_version/version.go [hunk 3]
```diff
@@ -18,21 +18,62 @@ package main
...
 )
 
 // Before: prev -> current (latest)
 // After:  prev -> current -> new (latest)
 func main() {
+	logrus.SetLevel(logrus.DebugLevel)
 	new := os.Args[1]
-	current := strings.TrimPrefix(schema.SchemaVersions[len(schema.SchemaVersions)-1].APIVersion, "skaffold/")
+	current := latest.Version
+	fmt.Printf("current version: %s", current)
```
Comment (dgageot) [inline hack/new_config_version/version.go]: `logrus.Infof`?

File: hack/new_config_version/version.go [hunk 4]
```diff
@@ -18,21 +18,62 @@ package main
...
+	"context"
+	"fmt"
 	"io/ioutil"
+	"net/http"
 	"os"
 	"path/filepath"
 	"regexp"
 	"strings"
 
+	"github.com/google/go-github/github"
```
Comment (dgageot) [inline hack/new_config_version/version.go]: goimports

File: hack/new_version.sh (hunk 1 above)
Comment (balopat) [reply hack/new_version.sh]: I'll add it in `version.go`

**Pull request merged**